azure/services/agentpools/agentpools.go

```diff
@@ -80,8 +81,9 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 	} else {
 		ps := *existingPool.ManagedClusterAgentPoolProfileProperties.ProvisioningState
 		if ps != "Canceled" && ps != "Failed" && ps != "Succeeded" {
```

Comment:
these should use the azure consts

Reply:
fixed + added "Canceled" as a terminal state